### PR TITLE
Add IOSMessaging module for APNS specific methods

### LINKED ISSUE #1511: Registering for notifications is coupled with the modal requestPermission
### Issue

Please expose a method other than RequestPermission to perform the wiring of APNS to FCMToken.  

The magic seems to be this call in RequestPermission:
registerForRemoteNotifications

The importance of this function was explained in the documentation when we started.  Right now the only way to perform this sync is to call requestPermission every time the fcmToken changes.  The code doesn't read clearly (I'm not really requestingPermission).  

It also created problems for us because we are trying to preserve our one chance on iOS to request permission, so our solutions to rewire existing users after token refresh also needs to skip requestPermission.

We'd like a second function called resyncForRemoteNotifications or something similar to perform only the sync operations.

### Environment
1. Application Target Platform:
Both

2. Development Operating System:
mac High Sierra

3. Build Tools:
xcode 9.4, Android Studio 28

4. React Native version:

"react-native": "0.55.4",

5. React Native Firebase Version:

"react-native-firebase": "^4.0.3",

6.  Firebase Module:

firebase-core:12.0.1
firebase-analytics:12.0.1
firebase-messaging:12.0.1

7. Are you using typescript?
no

### PR DESCRIPTION
**Pull request by timwangdev**
### Summary

Fixes #1511

The user should have the freedom when to call the registerForRemoteNotifications from application instance as described in firebase doc https://firebase.google.com/docs/cloud-messaging/ios/client#register_for_remote_notifications

And expose APNS token is very handy for debugging device token sync issue in iOS. This is not a standard firebase method.

### Checklist

- [x] Supports `Android`
- [x] Supports `iOS`
- [ ] `e2e` tests added or updated in [/tests/e2e/\*](/tests/e2e)
- [x] Updated the documentation in the [docs repo](https://github.com/invertase/react-native-firebase-docs)
  - invertase/react-native-firebase-docs#137
- [x] Flow types updated
- [x] Typescript types updated

### Test Plan

N/A

### Release Plan

[IOS] [FEATURE] [MESSAGING] - Add IOSMessaging module for APNS specific methods.

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

**Comment by timwangdev** [comment]
I will add the doc if this passed the initial review.

File: android/src/main/java/io/invertase/firebase/messaging/RNFirebaseMessaging.java [hunk 1]
```diff
@@ -64,6 +64,11 @@ public void requestPermission(Promise promise) {
     promise.resolve(null);
   }
 
+  @ReactMethod
+  public void registerForRemoteNotifications(Promise promise) {
```
Comment (Salakar) [inline android/src/main/java/io/invertase/firebase/messaging/RNFirebaseMessaging.java]: We can remove this and add an if (isAndroid) return null; line in the JS method, native code changes not required then 👍

File: src/modules/messaging/index.js [hunk 2]
```diff
@@ -137,6 +137,10 @@ export default class Messaging extends ModuleBase {
     return getNativeModule(this).hasPermission();
   }
 
+  registerForRemoteNotifications(): Promise<void> {
```
Comment (Salakar) [inline src/modules/messaging/index.js]: Could we namespace this under ios? It's good to keep platform-specific methods outside of the main cross-platform methods.

e.g. this becomes `firebase.messaging().ios.registerForRemoteNotifications()`

**Comment by timwangdev** [comment]
@Salakar I just merged #1625 to this branch since the function is similar and shared the same structure.

**Comment by timwangdev** [comment]
Added doc at https://github.com/invertase/react-native-firebase-docs/pull/137

**Comment by Salakar** [comment]
@timwangdev thanks for sorting the docs, and this PR, nice 👍 will merge now 🎉

**Pull request merged**